lib/scope.js

```diff
@@ -841,116 +850,182 @@ module.exports = {
     for ( let guess of field.guesses ) {
-      for ( let var_datum of var_data ) {
-        let row_matches = false;
+      for ( let table_row of var_data ) {
+        let this_row_matches_this_field = false;
 
         // If the dev uses single instead of double quotes, etc, allow that
-        let names_match = await scope.matchNeutralizedQuotes( scope, { value1: guess.var, value2: var_datum.var });
-        let values_match = await scope.matchNeutralizedQuotes( scope, { value1: guess.value, value2: var_datum.value });
+        let names_match = await scope.matchNeutralizedQuotes( scope, { value1: guess.var, value2: table_row.var });
+        let values_match = await scope.matchNeutralizedQuotes( scope, { value1: guess.value, value2: table_row.value });
 
         // Check different things for different fields. An input box won't
         // be expected to have the same value as the var you want to set while
         // a radio button and its table row need to have a matching value.
         switch ( field.tag ) {
           case 'button':
           case 'select':
-            if ( names_match && values_match ) { row_matches = true; }
+            if ( names_match && values_match ) { this_row_matches_this_field = true; }
             break;
 
           case 'canvas':
           case 'textarea':
           case 'a':
-            if ( names_match ) { row_matches = true; }
+            if ( names_match ) { this_row_matches_this_field = true; }
             break;
 
           case 'input':
             if ( field.type === 'radio' ) {
-              if ( names_match && values_match ) { row_matches = true; }
+              if ( names_match && values_match ) { this_row_matches_this_field = true; }
             } else {  // Some kind of text input
-              if ( names_match ) { row_matches = true; }
+              if ( names_match ) { this_row_matches_this_field = true; }
             }
             break;
         }
 
-        if ( row_matches ) {
-          matches.push( var_datum );  // Collect all matches to start with.
-          if ( !uses_proxies ) {  // If this gets set to `true` once, leave it alone
-            // x, i, j, etc. See notes below about the reasoning
-            // AVOID `.test()` - it advances through a string each time you call it on the
-            // string, so you can have unexpected results.
-            uses_proxies = guess.var.match( proxies_regex );  // 
+        if ( this_row_matches_this_field ) {
+          rows_that_match_field_names.push( table_row );
+          // Check this until the value is set to `true` (if it ever is)
+          if ( !uses_proxies ) {
+            // If the field uses proxy variables - x, i, j, etc, flag it. See "proxies" note.
+            uses_proxies = guess.var.match( proxies_regex );
           }
-          continue;
-        }
+
+          let table_row_trigger_name = table_row.trigger;
+          if ( table_row_trigger_name === `ALKiln: no trigger variable needed` ) {
```

Comment:
IMO no need to define a variable with a longer name than the original variable that's only used once.
```suggestion
          if ( table_row.trigger === `ALKiln: no trigger variable needed` ) {
```

Reply:
This was actually part of an effort I forgot about as I was going - a intermediate step for improving `.trigger` to `.trigger_name`.

Reply:
Maybe I should save it for a future, more extensive, PR.